### DISCUSSION TITLE
Consider providing source maps

### DISCUSSION THREAD

**Discussion by joshkel** [vercel/swr#1636] [Ideas] [answered]
Would you consider providing source maps for SWR's NPM package? (See #1079.) I see where they were removed, but there have been a few times where I'm using a debugger to step through my code to try to understand when and how data is being revalidated, and the minified no-source-map SWR source code is mostly a black box.

(If I'm understanding correctly, for this to be of benefit, the source maps would either have to include `sourcesContent` - I'm not sure if Bunchee provides this option - or the NPM package would have to include the original sources.)

Alternatively, the SWR NPM package could include unminified code - since its users are almost always going to be using a minifier of their own, I'm not sure how much it gains by being minified itself. (I could be wrong.)

**Comment by shuding** [comment, marked as answer]
Thanks for the feedback! The latest beta release now includes non-minified source code: https://unpkg.com/swr@1.1.0-beta.10/dist/index.js, hope it helps!
